world-explore/src/app/home/home.page.ts

```diff
@@ -66,16 +75,10 @@ export class HomePage implements OnInit {
   }
 
   filterCountries() {
-    if (this.showFavorites) {
-      this.filteredCountries = this.countries.filter(country =>
-        this.favoriteCountries.has(country.country) &&
-        country.country.toLowerCase().includes(this.searchTerm.toLowerCase())
-      );
-    } else {
-      this.filteredCountries = this.countries.filter(country =>
-        country.country.toLowerCase().includes(this.searchTerm.toLowerCase())
-      );
-    }
+    this.filteredCountries = this.countries.filter(country => 
```

Comment:
1- use toLowerCase() method: By converting both country.country and this.searchTerm to lowercase once, we ensure case-insensitive comparison without repeated method calls.
2- use Short-circuit evaluation: The expression (!this.showFavorites || this.favoriteCountries.has(country.country)) ensures that if this.showFavorites is false, the has() check won't be evaluated unnecessarily, improving performance.